interfaces/builtin/basedeclaration.go

```diff
@@ -190,6 +193,11 @@ slots:
       slot-snap-type:
         - core
     deny-auto-connection: true
+  classic-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
```

Comment:
I may definitely be missing something, but I'd appreciate some insight into what it is. If this is allowed into the core slot, then arbitrary snaps will be able to connect to it, won't they? Because the search logic is "plug, slot, base plug, base slot". We don't want any snaps other than "classic" (_the_ classic one, not other classic-confined ones) connecting to this slot.